### PR TITLE
Enable ESLint for TypeScript files

### PR DESCRIPTION
**Pull request by ThijsTyZ**

### CODE REVIEW

File: src/app/bootstrap.dev.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ const replaceVariables: { [index: string]: any } =
   ) || {};
 
 // bootstrap the app
-const appElement = document.getElementById('app');
+const appElement = document.querySelector('#app') as HTMLElement;
```
Comment (ThaNarie) [inline src/app/bootstrap.dev.ts:24]:
```suggestion
const appElement = document.querySelector<HTMLElement>('#app');
```

File: src/app/bootstrap.dist.ts [hunk 2]
```diff
@@ -32,7 +32,7 @@ context.keys().forEach(key => {
 });
 
 // Makes the website interactive
-const appElement = document.getElementById('app');
+const appElement = document.querySelector('#app') as HTMLElement;
```
Comment (ThaNarie) [inline src/app/bootstrap.dist.ts:35]:
```suggestion
const appElement = document.querySelector<HTMLElement>('#app');
```

File: src/app/component/block/paragraph/Paragraph.ts [hunk 3]
```diff
@@ -1,29 +1,30 @@
...
+    super(element);
 
     this.button = this.element.querySelector('button');
     if (this.button) {
       this.button.addEventListener('click', this.onButtonClick);
       this.buttonLabel = this.getElement<HTMLSpanElement>('.label', this.button);
       this.buttonIcon = this.getElement<HTMLElement>('[data-component="icon"]', this.button);
 
-      this.contentMore = <HTMLParagraphElement>this.element.querySelector('.js-content-more');
+      this.contentMore = this.element.querySelector('.js-content-more') as HTMLParagraphElement;
```
Comment (ThaNarie) [inline src/app/component/block/paragraph/Paragraph.ts:23]:
```suggestion
      this.contentMore = this.element.querySelector<HTMLParagraphElement>('.js-content-more');
```

File: src/app/muban/knockoutUtils.ts [hunk 4]
```diff
@@ -10,7 +10,7 @@ import extract from 'html-extract-data';
  */
 export function initTextBinding(element: HTMLElement, html: boolean = false): Observable<string> {
   // init the observable with the correct initial data
-  const obs = ko.observable(<string>element[html ? 'innerHTML' : 'textContent']);
+  const obs = ko.observable(element[html ? 'innerHTML' : 'textContent'] as string);
```
Comment (ThaNarie) [inline src/app/muban/knockoutUtils.ts:13]: The returned value could be `null`, so `?? ''` is safer.

```suggestion
  const obs = ko.observable(element[html ? 'innerHTML' : 'textContent'] ?? '');
```

**Review by ThaNarie — CHANGES_REQUESTED** [review]

**Review by ThaNarie — APPROVED** [review]

**Comment by leroykorterink** [comment]
We should also update the `block` and `smart-component` TypeScript templates.

Block:
```typescript
import AbstractBlock from '../AbstractBlock';

export default class {{name_pc}} extends AbstractBlock {
  public static readonly displayName = '{{name_sc}}';

  public constructor(element:HTMLElement) {
    super(element);
  }

  public dispose() {
    super.dispose();
  }
}
```

SmartComponent:
```typescript
import AbstractComponent from '../AbstractComponent';

export default class {{name_pc}} extends AbstractComponent {
  public static readonly displayName = '{{name_sc}}';

  public constructor(element: HTMLElement) {
    super(element);
  }

  public dispose() {
    super.dispose();
  }
}
```

**Pull request merged**